helper/schema/resource_data.go

```diff
@@ -171,6 +172,14 @@ func (d *ResourceData) SetPartial(k string) {
 	}
 }
 
+func (d *ResourceData) MarkNewResource() {
+	d.isNew = true
+}
+
+func (d *ResourceData) IsNewResource() bool {
+	return d.isNew
+}
```

Comment:
That is a good point. Making `IsNewResource()` just an "alias" for `Id() == ""` is much cleaner approach. I will update the PR.